cmake/compiler.cmake

```diff
@@ -211,6 +211,26 @@ option(ENABLE_WERROR "Make all compiler warnings into errors" OFF)
 
 option(ENABLE_UB_SANITIZER "Make the compiler generate runtime code to perform undefined behaviour checks" OFF)
 
+# _FORTIFY_SOURCE requires compiling with optimization
+if (CMAKE_BUILD_TYPE STREQUAL "Debug" OR
+    # LuaJIT in FreeBSD doesn't work with PIC (gh-7640)
+    TARGET_OS_FREEBSD)
+    set(ENABLE_HARDENING_DEFAULT FALSE)
+else()
+    set(ENABLE_HARDENING_DEFAULT TRUE)
+endif()
+option(ENABLE_HARDENING "Enable compiler options that harden against memory corruption attacks" ${ENABLE_HARDENING_DEFAULT})
```

Comment:
BTW, I would show it in the cmake output, similarly to 1546a1239be27a9b1e03f228b6c8be891133fb4b.

Reply:
Done